### PR TITLE
ADD: compatible with scala-2.11

### PR DESCRIPTION
**Pull request by xuanyu66**
### What problem does this PR solve? 

Make TiSpark support spark-2.3 by compatible with scala-2.11

### What is changed and how it works?

- use profile to change properties `scala.version` and `spark.version`
- move TiDDLRule and TiResolutionRule to spark-wrapper
- version with suffix `-scala_${scala.binary.version}`

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

**Comment by xuanyu66** [comment]
@sunxiaoguang @birdstorm  PTAL
https://ci2.pingcap.net/blue/organizations/jenkins/tispark_ghpr_integration_test/detail/tispark_ghpr_integration_test/173/pipeline
https://ci2.pingcap.net/blue/organizations/jenkins/tispark_ghpr_integration_test/detail/tispark_ghpr_integration_test/171/pipeline

**Review by birdstorm — DISMISSED** [review]
LGTM

File: core/src/main/scala/com/pingcap/tispark/TiSparkInfo.scala [hunk 1]
```diff
@@ -44,5 +44,9 @@ object TiSparkInfo {
       logger.error("Current TiSpark Version is not compatible with current Spark Version!")
       throw new TiInternalException("")
     }
+    if (SPARK_MAJOR_VERSION == "2.3" && util.Properties.versionNumberString.startsWith("2.12")) {
+      logger.error("Spark-2.3 is not compatible with scala-2.12, please use version ${TiSpark.version}-scala_2.11")
+      throw new TiInternalException("")
```
Comment (sunxiaoguang) [inline core/src/main/scala/com/pingcap/tispark/TiSparkInfo.scala:49]: How about use this error for the message of TiInternalException?

**Review by sunxiaoguang — APPROVED** [review]

**Pull request merged**